### PR TITLE
move boilerplate FunctionN into akka.japi.function package

### PR DESCRIPTION
**Pull request by rkuhn**
This is a temporary solution until the classes are added to akka-actor.

### CODE REVIEW

File: akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/io/StreamTcpDocTest.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
  */
-package docs.stream;
+package docs.stream.io;
```
Comment (ktoso) [inline akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/io/StreamTcpDocTest.java]: good catch

File: project/AkkaBuild.scala [hunk 2]
```diff
@@ -1436,7 +1436,7 @@ object AkkaBuild extends Build {
 
     // Temporary fix for #15379. Should be removed when stream is stabilized.
     // And yes OSGi wont like you mixing the persistence and stream artifacts.
-    val stream = exports(Seq("akka.stream.*", "akka.persistence.stream.*"))
+    val stream = exports(Seq("akka.stream.*", "akka.persistence.stream.*", "akka.japi.function.*")) //FIXME when akka.japi is fully in akka-actor
```
Comment (patriknw) [inline project/AkkaBuild.scala]: we don't have any persistence in here any more, but that doesn't matter

**Comment by patriknw** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM

**Comment by 2m** [comment]
LGTM

**Comment by drewhk** [comment]
LGTM

**Pull request merged**